curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64dc985e6720a23edac63b51.md

```diff
@@ -0,0 +1,51 @@
+---
+id: 64dc985e6720a23edac63b51
+title: Step 17
+challengeType: 20
+dashedName: step-17
+---
+
+# --description--
+
+The allowed disk movements between the rods exhibit a repetitive pattern occurring every three moves. For example, movements between rod `A` and rod `C` are allowed on the first, the fourth and the seventh move, where the remainder of the division between the move number and 3 is equal to 1.
+
+Inside the previously created `for` loop, replace the existing `print()` call with an `if` statement that is triggered when `(i + 1) % 3 == 1`. Within this `if` statement, print `f'Move {i + 1} allowed between {source} and {target}'` using an f-string. Please, note the use of `i + 1` as the move number, as `i` is zero during the first iteration.
```

Comment:
```suggestion
Inside the previously created `for` loop, replace the existing `print()` call with an `if` statement that is triggered when `(i + 1) % 3 == 1`. Within this `if` statement, print `f'Move {i + 1} allowed between {source} and {target}'` using an f-string. Please, note that `i + 1` is the move number since `i` is zero during the first iteration.
```